src/platform/plugins/shared/inspector/common/adapters/request/request_adapter.ts

```diff
@@ -61,14 +73,24 @@ export class RequestAdapter extends EventEmitter {
   }
 
   public resetRequest(id: string): void {
+    const req = this.requests.get(id);
     this.requests.delete(id);
+    if (req) {
+      this.responses.delete(req);
+    }
     this._onChange();
   }
 
   public getRequests(): Request[] {
     return Array.from(this.requests.values());
   }
 
+  public getRequestEntries(): Array<[Request, RequestResponder]> {
+    return this.getRequests()
+      .map((req) => [req, this.responses.get(req)] as [Request, RequestResponder])
+      .filter(([_req, responder]) => responder != null);
+  }
```

Comment:
nit: you can move the `as [Request, RequestResponder]` in the return of the filter, because at the casting this in the map callback doesn't make much sense (is filter that produce an array with both defined values, not the map.

You can also use a flatMap instead without the need of a cast:
```
return this.getRequests().flatMap((req) => {
      const responder = this.responses.get(req);
      return responder ? [[req, responder]] : [];
    });
```

Reply:
The `filter` is there to prevent partial or failed request to land on the adapter.
As for the `flatMap` approach, it's a subjective topic.
I prefer the `filter` as it pushed less memory pressure in this case ( 1 extra array generated ) vs the `flatMap` ( N arrays generated ).
Not a big deal, but if it's not strictly necessarily I would avoid it.